core/src/syscall/precompiles/blake2b/compress/air.rs

```diff
@@ -0,0 +1,248 @@
+use std::borrow::Borrow;
+
+use crate::{
+    air::{BaseAirBuilder, SP1AirBuilder, WORD_SIZE},
+    syscall::precompiles::blake2b::{MSG_ELE_PER_CALL, STATE_ELE_PER_CALL},
+};
+use p3_air::{Air, AirBuilder, BaseAir};
+use p3_field::AbstractField;
+use p3_matrix::MatrixRowSlices;
+
+use super::{
+    columns::{Blake2bCompressInnerCols, NUM_BLAKE2B_COMPRESS_INNER_COLS},
+    mix::MixOperation,
+    Blake2bCompressInnerChip, MIX_INDEX, NUM_MIX_ROUNDS, NUM_STATE_WORDS_PER_CALL, OPERATION_COUNT,
+    SIGMA_PERMUTATIONS,
+};
+
+impl<F> BaseAir<F> for Blake2bCompressInnerChip {
+    fn width(&self) -> usize {
+        NUM_BLAKE2B_COMPRESS_INNER_COLS
+    }
+}
+
+impl<AB> Air<AB> for Blake2bCompressInnerChip
+where
+    AB: SP1AirBuilder,
+{
+    fn eval(&self, builder: &mut AB) {
+        let main = builder.main();
+        let local: &Blake2bCompressInnerCols<AB::Var> = main.row_slice(0).borrow();
+        let next: &Blake2bCompressInnerCols<AB::Var> = main.row_slice(1).borrow();
+
+        self.constrain_control_flow_flags(builder, local, next);
+
+        self.constrain_memory(builder, local);
+
+        self.constrain_mix_operation(builder, local);
+    }
+}
+
+impl Blake2bCompressInnerChip {
+    /// Constrains the given index is correct for the given selector. The `selector` is an
+    /// `n`-dimensional boolean array whose `i`-th element is true if and only if the index is `i`.
+    fn constrain_index_selector<AB: SP1AirBuilder>(
```

Comment:
do we not already have a generic thing of doing this? this seems like a really good utility to have in airbuilder in general tbh, so maybe we should move it to a move general crate with all the existing air builder methods

Reply:
should I address this issue in the current PR, or should it be handled in a separate PR by creating an issue?